### PR TITLE
Add marketing info banner to gifting checkout page

### PR DESCRIPTION
**Pull request by DustyReagan**
#### Proposed Changes

* This PR adds a marketing info banner to the Gifts checkout page.
* The intention is to increase checkout conversions by providing more information and tying the checkout page back to the original Gifting banner, so the user feels like they're in the right place.
* We try to load the banner quickly to reduce user bouncing.

Before | After
--|--
gift-order-before  |  gift-order-after

#### Testing Instructions

* Load this PR
* Make sure it the Gifting checkout order banner only displays on Gifting checkouts. It should not appear on any other checkout pages.
* Make sure the Gifting checkout banner is responsive and looks good on mobile.
* Makes sure the Gifting banner CSS is specific to our banner and won't affect other areas of the site.

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to https://github.com/Automattic/dotcom-forge/issues/1422

### CODE REVIEW

File: client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx [hunk 1]
```diff
@@ -298,166 +299,171 @@ export default function WPCheckout( {
...
-					<WPCheckoutOrderReview
-						forceRadioButtons={ forceRadioButtons }
-						removeProductFromCart={ removeProductFromCart }
-						couponFieldStateProps={ couponFieldStateProps }
-						onChangePlanLength={ changePlanLength }
-						siteUrl={ siteUrl }
-						createUserAndSiteBeforeTransaction={ createUserAndSiteBeforeTransaction }
-					/>
+		<>
+			<CheckoutOrderBanner />
```
Comment (DustyReagan) [inline client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx]: I only added `<CheckoutOrderBanner />` and wrapped everything in `<></>` so it has a parent element. I also added a [new className](https://github.com/Automattic/wp-calypso/pull/71823/files#r1068788484). The rest of the changes below are indentation due to the new `<>` parent element.

File: client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx [hunk 2]
```diff
@@ -298,166 +299,171 @@ export default function WPCheckout( {
...
+								<CheckoutSummaryTitle>
+									<CheckoutSummaryTitleIcon icon="info-outline" size={ 20 } />
+									{ translate( 'Purchase Details' ) }
+									<CheckoutSummaryTitleToggle icon="keyboard_arrow_down" />
+								</CheckoutSummaryTitle>
+								<CheckoutSummaryTitlePrice className="wp-checkout__total-price">
+									{ total.amount.displayValue }
+								</CheckoutSummaryTitlePrice>
+							</CheckoutSummaryTitleLink>
+							<CheckoutSummaryBody className="checkout__summary-body">
```
Comment (DustyReagan) [inline client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx:321]: I added a new className here to target CSS off of.

File: client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx (hunk 1 above)
Comment (DustyReagan) [reply client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx]: I moved `<CheckoutOrderBanner />` up the stack so this is no longer accurate.

File: client/my-sites/checkout/composite-checkout/components/checkout-order-banner/gifting-checkout-banner/style.scss [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+body .gifting-checkout-banner + .composite-checkout .checkout__summary-area {
+	position: -webkit-sticky; /* Legacy Browsers */
+	position: sticky;
+	top: 79px;
+}
+body .gifting-checkout-banner + .composite-checkout .checkout__summary-body {
+	position: initial;
+	height: initial;
+}
```
Comment (DustyReagan) [inline client/my-sites/checkout/composite-checkout/components/checkout-order-banner/gifting-checkout-banner/style.scss:67]: This sticks the right hand checkout summary box to the top of the page so it behaves like checkout pages without the Gifting banner.

File: client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx [hunk 4]
```diff
@@ -315,7 +315,7 @@ export default function WPCheckout( {
 								{ total.amount.displayValue }
 							</CheckoutSummaryTitlePrice>
 						</CheckoutSummaryTitleLink>
-						<CheckoutSummaryBody>
+						<CheckoutSummaryBody className="checkout__summary-body">
```
Comment (DustyReagan) [inline client/my-sites/checkout/composite-checkout/components/wp-checkout.tsx:318]: I added the new `checkout__summary-body` className here to target CSS off of.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7732139

Thank you @DustyReagan for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/checkout/composite-checkout/components/checkout-order-banner/index.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { useShoppingCart } from '@automattic/shopping-cart';
+import * as React from 'react';
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/components/checkout-order-banner/index.tsx]: Minor: you shouldn't need this import. React is imported automatically by our build system.

**Review by sirbrillig — APPROVED** [review]
You should get a review from your own team as well, but from a checkout perspective this looks good to me and works as described at both desktop and mobile widths.

Screenshot 2023-01-13 at 3 35 50 PM

Screenshot 2023-01-13 at 3 35 42 PM

File: client/my-sites/checkout/composite-checkout/components/checkout-order-banner/index.tsx (hunk 5 above)
Comment (DustyReagan) [reply client/my-sites/checkout/composite-checkout/components/checkout-order-banner/index.tsx]: Removed. Thanks! ☺️

**Comment by JanaMW27** [comment]
Hello! Could we introduce the banner in the congrats page to create visual continuity?
Something like:
Congrats - marketplace

Do we need a new PR for this?
Thanks!

**Comment by DustyReagan** [comment]
> Hello! Could we introduce the banner in the congrats page to create visual continuity? Something like: ...
> 
> Do we need a new PR for this? Thanks!

Good idea, @JanaMW27! It should be a new PR. I created a [new issue for it here](https://github.com/Automattic/dotcom-forge/issues/1505) so we can track it.

**Review by paulopmt1 — COMMENTED** [review]
It's working as expected @DustyReagan!

Should we use replace the current fixed `background-size` with `background-size: cover;` so we can have better spacing in this mobile scenario?

image

**Comment by DustyReagan** [comment]
> Should we use replace the current fixed `background-size` with `background-size: cover;` so we can have better spacing in this mobile scenario?

Thanks for pointing this out, @paulopmt1!

I just pushed an update to address this as well as extra long domain names that create 3 lines of copy. I did this by separating the image of a woman from the pink background. Please let me know what you think! ☺️

https://user-images.githubusercontent.com/140841/213024757-f00aad5b-42a2-4578-aa47-dfd2512e97b1.mp4

**Review by paulopmt1 — APPROVED** [review]
Great job @DustyReagan! Loved your solution!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7732139).

**Pull request merged**